### PR TITLE
Improve search accessibility

### LINKED ISSUE #5948: [A11Y] Search bar/box don't have accessible labels
**Describe the bug**

Search bar only has a placeholder but no label.

**Additional context**

- https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions

### LINKED ISSUE #5949: [A11Y] Status messages are not announced in the search app
**Describe the bug**

When searching, status messages (e.g., "No result") are not announced by the screen reader.

**To Reproduce**
1. Go to the search page with screen reader turned on
2. Type a bogus keyword into the search field
3. Observe that the "No result" message is shown but screen reader does not announce it

**Expected behavior**

Screen reader announces the status messages.

**Additional context**

- https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html
- https://talk.getindico.io/t/status-message-search-app-live-notices/3274

### PR DESCRIPTION
**Pull request by foxbunny**
- Add proper labels to SearchBar and SearchBox
- Make screen readers announce the status messages

Fixes #5948, fixes #5949

### CODE REVIEW

File: indico/modules/search/client/js/components/SearchApp.jsx [hunk 1]
```diff
@@ -330,9 +330,11 @@ export default function SearchApp({category, eventId, isAdmin}) {
           sortOptions={options?.sortOptions || []}
           onSortChange={value => handleQuery(value, 'sort')}
         />
-        {!isAnyLoading && (
-          <ResultHeader query={q} hasResults={!!results.total} categoryTitle={category?.title} />
-        )}
+        <div aria-live="polite">
```
Comment (ThiefMaster) [inline indico/modules/search/client/js/components/SearchApp.jsx:333]: should the div be there all the time or only when the header is shown?

File: indico/modules/search/client/js/components/SearchBar.module.scss [hunk 2]
```diff
@@ -15,3 +16,7 @@
     margin-right: 1em !important;
   }
 }
+
+.searchField > span {
```
Comment (ThiefMaster) [inline indico/modules/search/client/js/components/SearchBar.module.scss:20]: Just wondering, wouldn't it be better to have a global `visually-hidden` class for this, and put it directly on the span element? Before seeing this I was wondering why the span is necessary at all...

File: indico/modules/search/client/js/components/SearchApp.jsx (hunk 1 above)
Comment (foxbunny) [reply indico/modules/search/client/js/components/SearchApp.jsx:333]: It must be present at all times.

File: indico/modules/search/client/js/components/SearchBar.module.scss (hunk 2 above)
Comment (foxbunny) [reply indico/modules/search/client/js/components/SearchBar.module.scss:20]: I don't like to address styling concerns in the markup. Markup is not intended to be used for styling.

Comment (foxbunny) [reply indico/modules/search/client/js/components/SearchBar.module.scss:20]: (A.k.a. "Good ol' separation of concerns", that's how I was taught.)

**Comment by foxbunny** [comment]
I just noticed I use the wrong style for the class name (camelCase instead of kebab-case).

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**